temporal/fx.go

```diff
@@ -132,6 +132,10 @@ type (
 )
 
 func NewServerFx(opts ...ServerOption) *ServerFx {
+	var fxOpts []fx.Option
+	for _, opt := range opts {
+		fxOpts = append(fxOpts, opt.fxOptions()...)
+	}
```

Comment:
why do we need to make a copy? Can we use opt.fxOptions()... directly to the call of fx.New()

Reply:
Because there could be multiple ServerOptions which provide `fxOptions()` in the `opts` array.

Reply:
This is basically flattening a list. It doesn't copy the actual objects, just pointers to them